lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -282,97 +398,167 @@ def parse_genbank(self, file_path, params):
-                        unmatched_ids_md5s.append(current_contig)
-                else:
-                    unmatched_ids.append(current_contig)
-            if len(unmatched_ids) > 0:
-                raise ValueError(warnings['assembly_ref_extra_contigs'].format(", ".join(unmatched_ids)))
-            if len(unmatched_ids_md5s) > 0:
-                raise ValueError(warnings["assembly_ref_diff_seq"].format(", ".join(unmatched_ids_md5s)))
-            logging.info(f"Using supplied assembly: {assembly_ref}")
-            return assembly_ref
-        logging.info("Saving sequence as Assembly object")
-        Bio.SeqIO.write(out_contigs, fasta_file, "fasta")
-        assembly_ref = self.aUtil.save_assembly_from_fasta(
-            {'file': {'path': fasta_file},
-             'workspace_name': params['workspace_name'],
-             'assembly_name': assembly_id,
-             'type': params.get('genome_type', 'isolate'),
-             'contig_info': extra_info})
-        logging.info(f"Assembly saved to {assembly_ref}")
-        return assembly_ref
+            genome_obj.contig_seq[in_contig.id] = in_contig.seq.upper()
+        return out_contigs, extra_info
+
+    def _get_contigs_and_validate_existing_assembly(self, genome_objs):
+        ref2genome = {}
+        for genome_obj in genome_objs:
+            assembly_ref = genome_obj.use_existing_assembly
+            if assembly_ref:
+                contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
+                _, extra_info = self._get_contigs_and_extra_info(
+                    contigs, genome_obj
+                )
+                self._validate_existing_assembly(
+                    assembly_ref, genome_obj
+                )
+                ref2genome[assembly_ref] = genome_obj
+                genome_obj.extra_info = extra_info
+                genome_obj.assembly_ref = assembly_ref
+                genome_obj.assembly_path = None
+
+        if ref2genome:
+            assembly_refs = list(ref2genome.keys())
+            results = self._get_objects_data(assembly_refs)
+            for assembly_ref, result in zip(assembly_refs, results):
+                object_data = result["data"]
+                assembly_info = result["info"]
+                genome = ref2genome[assembly_ref]
+                genome.assembly_info = assembly_info
+                genome.gc_content = object_data["gc_content"]
+                genome.dna_size = object_data["dna_size"]
+                genome.md5 = object_data["md5"]
+
+    def _save_assemblies(self, workspace_id, genome_objs):
+        id2genome = {}
+        bulk_inputs = []
+        for genome_obj in genome_objs:
+            if genome_obj.assembly_ref:
+                continue
+            contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
+            genome_obj.assembly_id = f"{genome_obj.genome_name}_assembly"
+            genome_obj.assembly_path = f"{self.cfg.sharedFolder}/{genome_obj.assembly_id}.fasta"
```

Comment:
is there a specific reason we are using `.fasta` instead of `.fa` extension here? I think by default AssemblyUtil.get_assembly_as_fasta returns `.fa` extension. This is also what collections downloader expects. 
https://github.com/kbase/collections/blob/main/src/loaders/workspace_downloader/workspace_downloader.py#L148

Reply:
This is from old code.
I don't mind switching to .fa extension and hope it will not break anything.

Reply:
If it's the way it's always worked we should just leave it alone to reduce the chance of breaking other code

Unless there's a really good reason to take that risk

Reply:
okay. We can leave it as `.fasta` from GFU. We probably need to work around and rename the file from collection's side. It's breaking some of collection's assumption. 
For example, 
https://github.com/kbase/collections/blob/main/src/loaders/common/loader_helper.py#L331-L348

Reply:
ah, but why that's working beforehand? Nothing has changed though.

Reply:
GFU hasn't been integrated into collections yet, Tian is working on it right now, so we're just starting to find integration issues, I assume

Reply:
We have never embedded GFU into collections. We only used AssemblyUtil and the default get_assembly_as_fasta from AssemblyUtil returns `.fa` files.

Reply:
jinx